### PR TITLE
Depend on packages instead of submodules

### LINKED ISSUE #2345: NuGet.Commands embedded copy of FileSystemGlobbing is now public
Because `preprocess` code is not available in RC2, the code that "internalizes" the copy of FileSystemGlobbing that NuGet.Commands embeds isn't in effect. As a result, the classes are public and conflict with the real FileSystemGlobbing. I've worked around it in CLI, but it should be fixed.

/cc @emgarten @yishaigalatzer @davidfowl

### PR DESCRIPTION
**Pull request by PatoBeltran**
## Bug

Fixes: https://github.com/NuGet/Home/issues/2345

## Fix

Remove the dependency on FileSystem and Common submodules to use the specific packages instead.

### CODE REVIEW

**Review by zivkan — APPROVED** [review]
I always disliked submodules, so yay for this PR! :)

File: src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: I have my reservations about this. 
Do we really need these?
No one else in the product brings them?

File: src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj]: Unless there are APIs are changes, why use the preview versions? 
Especially if we need to include these in the VSIX.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: Also you need to make sure these are signed if we are going to pack them in the VSIX.

**Review by nkolev92 — COMMENTED** [review]
I want to understand whether we absolutely have to add them to the VSIX. 

Note that we don't have many other dependencies in the VSIX.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude
Comment (PatoBeltran) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: @nkolev92 I just tested this. The three of them are used when adding contentFiles during restore. None of them are brought in by anything else, therefore if we want to use them we need to add them to the VSIX ourselves.

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: Ok, so we know what we'd need to do there. 

How about the dotnet.exe? 
Are they in the SDK already? Those packages would need to be source-build-able.

Comment (jainaashish) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: being part of asp.net I've no reason to believe that these packages aren't source-buildable.

**Review by jainaashish — APPROVED** [review]
Before merging this PR, run a private RPS and make sure there is no new regression because of new assemblies being added in the vsix.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude
Comment (PatoBeltran) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude:34]: They are not in the SDK already and they are also not source-build-able... There's a plan to add the in .NET Core 3.0

**Comment by nkolev92** [comment]
This PR has been on hold for too long. We can resurrect when necessary.

**Pull request closed without merging**
